### PR TITLE
CORE:  add endpoint GET: /api/topics - nc-news-2

### PR DESCRIPTION
**Pull request by jetakazono**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+describe("GET:/api/topics", () => {
+    test("status: 200 - should resolve with topics array ", () => {
+        return request(app)
+            .get("/api/topics")
+            .expect(200)
+            .then(({ body }) => {
+                const { topics } = body
+                expect(topics).toBeInstanceOf(Array)
+                expect(topics).toHaveLength(3)
+            })
```
Comment (Jodiech) [inline __tests__/app.test.js]: Nice to see you are checking the length here 👍

File: __tests__/app.test.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            .get("/api/topics")
+            .expect(200)
+            .then(({ body }) => {
+                const { topics } = body
+
+                topics.forEach((topic) => {
+                    expect(topic).toHaveProperty("slug"), expect.any(String)
+                    expect(topic).toHaveProperty("description"),
+                        expect.any(String)
+                })
```
Comment (Jodiech) [inline __tests__/app.test.js:24]: More a personal choice, but I'd argue it makes it cleaner if you were to condense these two tests together 😄

File: errors/errors.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+exports.handleCustomErrors = (err, req, res, next) => {
+    if (err.status && err.msg) {
+        res.status(err.status).send({ msg: err.msg })
+    } else next(err)
+}
+
```
Comment (Jodiech) [inline errors/errors.js]: We don't need to include PSQL error handling just yet, please remove this until it's needed 👍

File: errors/errors.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+exports.handlePsqlErrors = (err, req, res, next) => {
+    if (err.code === "22P02") {
+        res.status(400).send({ msg: "Invalid input" })
+    } else next(err)
+}
+
+exports.handleServerErrors = (err, req, res, next) => {
```
Comment (Jodiech) [inline errors/errors.js:1]: Nice error handling for a 500 here 🚀

File: app.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+const app = express()
+const apiRouter = require("./routes/api-route")
+
+const {
+    handleCustomErrors,
+    handlePsqlErrors,
+    handleServerErrors,
+} = require("./errors/errors")
+
+app.use(express.json())
```
Comment (Jodiech) [inline app.js]: we do not need to include this just yet, please remove it 😄

File: controllers/topics.controller.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
+const { selectAllTopics } = require("../models/topics.models")
+
+exports.getAllTopics = (_, res, next) => {
+    selectAllTopics()
+        .then((topics) => {
+            res.status(200).send({ topics })
+        })
+        .catch((err) => {
```
Comment (Jodiech) [inline controllers/topics.controller.js:8]: Good to see you have included a catch block here 👍

File: listen.js [hunk 7]
```diff
@@ -0,0 +1,6 @@
+const app = require("./app")
+
+app.listen(9090, (err) => {
+    if (err) console.log(err)
+    else console.log("app is listening on port 9090")
+})
```
Comment (Jodiech) [inline listen.js]: You won't need a listen.js file just yet as supertest will attach it's own listener. Please remove this 👍

**Review by Jodiech — COMMENTED** [review]
Hey 👋 nice work on this ticket! There are a few comments dotted around for you to take a look at. It's great that you have extracted your error handling however, there are a few things that are not needed at this stage. Please view my comments and make amendments. Please send another PR once this has been actioned 😄

**Review by pippaamy — APPROVED** [review]
great job on those changes, go ahead and merge :)

**Pull request merged**